src/client/src/components/homepage.js

```diff
@@ -0,0 +1,75 @@
+const React = require('react')
+const {Link, Redirect} = require('react-router-dom')
+const fetch = require('node-fetch')
+const logo = require('../logo.svg')
+const responseCodes = require('../statusCodeResponses')
+const LOGIN_URL = '/login'
+
+class HomePage extends React.Component {
+    constructor(props){
+        super(props)
+
+        this.state = {
+            username: '',
+            password: '',
+            successfulLogin: false
+        }
+
+        this.handleChange = this.handleChange.bind(this)
+        this.handleSubmit = this.handleSubmit.bind(this)
+    }
+
+    loginHandler(username, password){
+        fetch(LOGIN_URL, {
+            method: 'POST',
+            headers: { 'Content-Type': 'application/json' },
+            body: JSON.stringify({username, password}),
+            credentials: 'same-origin'
+        }).then(response => {
+            if(response.status === 200){
+                this.setState({successfulLogin: true})
+            }
+            if(responseCodes.hasOwnProperty(response.status)){
+                alert(responseCodes[response.status])
+                console.log(response.status, response)
+            }
+        }).catch(err => console.err(err))
+    }
+
+    handleChange(event){
+        const settings = {'Username': 'username', 'Password': 'password'}
+        const key = settings[event.target.placeholder]
+        this.setState({ [key] : event.target.value})
+    }
+    handleSubmit(event){
+        this.loginHandler(this.state.username, this.state.password)
+        event.preventDefault()
+    }
+    render(){
+        if(this.state.successfulLogin){
+            return <Redirect to={{
+                pathname: '/chat',
+                state: {username: this.state.username}
+            }}/>
+        }
+        return (
+            <div className="App">
+            <div className="App-header">
+                <img src={logo} className="App-logo" alt="logo" />
+                <h2>Welcome to Loki Instant Messenger</h2>
+                <p className="App-intro">Hello LokIM User!</p>
+            </div>
+            <form onSubmit={this.handleSubmit}>
+            <input className='userInput' type="text" value={this.state.username} onChange={this.handleChange} placeholder="Username" required/><br/>
+            <input className='userInput' type="password" value={this.state.password} onChange={this.handleChange} placeholder="Password" required/><br/>
+            <input type='submit' className='homeButton btn btn-primary' value='Login'/>
```

Comment:
`homeButton btn btn-primary` You mix camel case with kebab case here

Reply:
kebab case is bootstrap unfortunately, I am unable to change this and I am unwilling to move from JS standard of camelCase

Reply:
JS standard is not css standard. Let's keep things consistent and adapt kebab case here.